cmd/pinniped-proxy/src/pinniped.rs

```diff
@@ -152,6 +158,25 @@ async fn call_pinniped_exchange(authorization: &str, k8s_api_server_url: &str, k
         },
     }
 }
+    
+fn get_token_credential_request(client: Client, pinniped_namespace: &str) -> Api<TokenCredentialRequest> {
+    let ns_token_cred_req: bool = env::var(NAMESPACED_TOKEN_CREDENTIAL_REQUESTS).unwrap_or("false".into()).parse().unwrap();
```

Comment:
Note that this line can cause the app to panic, because you're calling `.unwrap()` which means to unwrap the result and panic if it's an error. More [info in the docs](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/the-question-mark-operator-for-easier-error-handling.html). What you should instead be doing is using the `?` operator which effectively says return an error if there is one, otherwise unwrap the result, and update the function return type to be a result
```
fn get_token_credential_request(client: Client, pinniped_namespace: &str) -> Result<Api<TokenCredentialRequest>> {
    let ns_token_cred_req: bool = env::var(NAMESPACED_TOKEN_CREDENTIAL_REQUESTS).unwrap_or("false".into()).parse()?;
...
    return Ok(Api::...);
}
```

Reply:
Mmm, interesting. However, I thought it wouldn't be required since `.unwrap_or()` fallback  to `false` and `"false".parse()` should always be `false`.

Reply:
Oh sure, yes if you are certain that the unwrap there can never result in an error. I'd not realised that as the case. Perfect as is then :)